src/proxy/connection_manager.rs

```diff
@@ -12,16 +12,31 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::identity::Identity;
 use crate::proxy::error;
 use crate::rbac::Connection;
 use crate::state::DemandProxyState;
 use drain;
+use http_types::convert::Serialize;
 use std::collections::HashMap;
+use std::net::SocketAddr;
 use std::sync::Arc;
 use tokio::sync::watch;
 use tokio::sync::RwLock;
 use tracing::info;
 
+#[derive(Clone, Copy, Hash, Debug, Eq, PartialEq)]
+pub struct ConnectionTuple {
+    pub src: SocketAddr,
+    pub dst: SocketAddr,
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct ConnectionMetadata {
+    #[serde(default)]
+    identity: Option<Identity>,
```

Comment:
I am confused about a connection's identity. Usually, we say the src identity or dest identity

Reply:
In the implementation we've got it will be the source identity. I suppose keeping the type general would allow us to represent peer identity generically with the same type in the future though.

Reply:
I documented a bit more now